WalletWasabi.Fluent/Helpers/MacOsStartupHelper.cs

```diff
@@ -8,22 +9,38 @@ namespace WalletWasabi.Fluent.Helpers;
 
 public static class MacOsStartupHelper
 {
-	private static readonly string AddCmd = $"osascript -e \'tell application \"System Events\" to make new login item at end with properties {{name:\"{Constants.AppName}\", path:\"/Applications/{Constants.AppName}.app\", hidden:true}}\'";
-	private static readonly string DeleteCmd = $"""osascript -e 'tell application "System Events" to delete login item "{Constants.AppName}"'""";
-	private static readonly string ListCmd = $"""osascript -e 'tell application "System Events" to get the name of every login item'""";
+	private static readonly string PlistContent =
+		$"""
+		<?xml version=\"1.0\" encoding=\"UTF-8\"?>
+		<!DOCTYPE plist PUBLIC \"-//Apple//DTD PLIST 1.0//EN\" \"http://www.apple.com/DTDs/PropertyList-1.0.dtd\">
+		<plist version=\"1.0\">
+		<dict>
+		    <key>Label</key>
+		    <string>com.wasabiwallet.startup</string>
+			<key>ProgramArguments</key>
+			<array>
+				<string>{EnvironmentHelpers.GetExecutablePath()}</string>
+				<string>{StartupHelper.SilentArgument}</string>
+			</array>
+			<key>RunAtLoad</key>
+			<true/>
+		</dict>
+		</plist>
+		""";
 
-	public static async Task AddOrRemoveLoginItemAsync(bool runOnSystemStartup)
+	public static async Task AddOrRemoveStartupItemAsync(bool runOnSystemStartup)
 	{
-		string result = await EnvironmentHelpers.ShellExecAndGetResultAsync(ListCmd).ConfigureAwait(false);
-		bool loginItemExists = result.Contains(Constants.AppName);
+		string homeDir = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+		string plistPath = Path.Combine(homeDir, "Library/LaunchAgents/", Constants.SilentPlistName);
 
-		if (!loginItemExists && runOnSystemStartup)
+		var fileExists = File.Exists(plistPath);
+		if (runOnSystemStartup)
```

Comment:
We write even if the file is there?

Reply:
it was explained in the other PR, but basically this is important to override the file if its content changed
Let's say that we launch binary built from source, next time we want to auto-launch the binary built from source
Let's say that we launch binary from released software, next time we want to auto-launch the binary from released software

If we don't override every time, the first time you launch this code will decide whether it will always launch release or whether it will always launch built from source.

Rewriting such small file is fairly cheap. If you prefer, we could also manage 2 different file names and check existence or delete files with a wildcard